### PR TITLE
feat: add telegraf filters and user guard

### PR DESCRIPTION
**Pull request by Archeri2000**
**Features**:

- Adds a user guard that can be used to guard telegram bot commands and ensure that the user has logged in.
- Change the start command to show what credentials you're logged in with
- Added logout command to clear the existing credentials

https://github.com/opengovsg/telegovsg/assets/16084793/e10162e5-7bf8-4653-be0d-ebe09c52fd92

**How to use**:
image
Using `@UseGuards(UserGuard)` will check prior to command execution that the user is authenticated, otherwise it will send the error message that the user is not authenticated and prompt them to login instead.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -5,5 +5,5 @@ BOT_TOKEN=yourbottoken
 # Set this to be the url given by ngrok
 BOT_DOMAIN=https://xxxx-xxx-xxx-xx-x.ngrok-free.app
 BOT_PATH=/webhook
-DATABASE_URL=postgres://pg.neon.tech:5432/neondb
-DATABASE_SSL_ENABLED=true
\ No newline at end of file
+DATABASE_URL=postgresql://root:root@localhost:5432/app
+DATABASE_SSL_ENABLED=false
```
Comment (LoneRifle) [inline .env.example:9]: Erm can we not? I don't really want to oblige the developer to figure out docker etc.

File: src/filters/telegraf-exception-filter.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@Catch(TelegrafException)
+export class TelegrafExceptionFilter implements ExceptionFilter {
+  private readonly logger = new Logger('TelegrafException');
+
+  async catch(exception: TelegrafException): Promise<void> {
+    this.logger.log(exception.message);
+    // const telegrafHost = TelegrafArgumentsHost.create(host);
+    // const ctx = telegrafHost.getContext<Context>();
+    //
+    // await ctx.replyWithHTML(`<b>Error</b>: ${exception.message}`);
```
Comment (LoneRifle) [inline src/filters/telegraf-exception-filter.ts:13]: ? If you think useful to the developer, perhaps add comments?

**Review by LoneRifle — CHANGES_REQUESTED** [review]
lgtm otherwise! One point of contention to discuss

File: .env.example (hunk 1 above)
Comment (Archeri2000) [reply .env.example:9]: Should I just remove the docker compose file and everything?

File: src/filters/telegraf-exception-filter.ts (hunk 2 above)
Comment (Archeri2000) [reply src/filters/telegraf-exception-filter.ts]: done!

File: .env.example (hunk 1 above)
Comment (LoneRifle) [reply .env.example:9]: you can keep it there, but perhaps remove the npm run commands and reference to localhost

Comment (Archeri2000) [reply .env.example:9]: ok!

Comment (Archeri2000) [reply .env.example]: done

**Review by LoneRifle — APPROVED** [review]
lgtm, nice work!

**Pull request merged**